src/Void.Core/CommandMode.fs

```diff
@@ -15,23 +16,40 @@ module CommandMode =
     let private requestLanguageForInterpreting buffer =
         buffer, GetCurrentCommandLanguageRequest :> Message
 
-    let handleGetCurrentCommandLanguageResponse buffer response =
+    let private interpretFragment buffer maybeResponse =
         {
-            Language = response.CurrentCommandLanguage
-            Fragment = !buffer
-        } :> Message
+            Language =

```

Comment:
Dude even better `maybe`:
http://hackage.haskell.org/package/base-4.8.1.0/docs/Prelude.html#v:maybe

`Hoogle` is so cool. I literally wrote down:
```
Maybe a -> (a -> b) -> b -> b
```

and it pointed me right to that puppy.